shell/gpu/gpu_surface_delegate.h

```diff
@@ -9,6 +9,23 @@ class GPUSurfaceDelegate {
  public:
   virtual ~GPUSurfaceDelegate() {}
 
+  //----------------------------------------------------------------------------
+  /// @brief      Gets the pixel geometry of the surface, i.e. whether the
+  ///             physical display is an LCD using BGR or RGB pixel layout and
+  ///             whether it is horizontally or vertically oriented.
+  ///             A change in this value will invalidate the current SkSurface.
+  ///             This value must be set to kUnknown_SkPixelGeometry in the
+  ///             case of raster backed surfaces such as a software surface.
+  ///
+  /// @return     The SkPixelGeometry of the current display the surface is
+  ///             rendered on. If the surface is being rendered on multiple
+  ///             displays with differing pixel geometry, it is up to the
+  ///             platform to pick which one to use.
+  ///
+  virtual SkPixelGeometry GetPixelGeometry() const {
```

Comment:
Nit: Virtual methods need to be OOL. Can you also move the destructor OOL while you are at it please?

Reply:
Done